### PR TITLE
Move to .NET 5 Preview 8 SDK

### PR DESCRIPTION
**Pull request by 333fred**
Moves us to .NET 5 Preview 8, including the P8 SCI and SRM binaries.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-infrastructure please take a look.

I've kicked of a signed build for a test insertion:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4014685&view=results

Test insertion:
<will be updated when created>

File: eng/Versions.props
Comment (333fred) [inline eng/Versions.props:246]: These are the public versions available on NuGet.

File: global.json
Comment (sharwell) [inline global.json:10]: ❔ This needs an update as well right?

Comment (333fred) [reply global.json:10]: Yeah, probably. I just want to see the CI work first :)

**Comment by jaredpar** [comment]
FYI: this will mean that we need to move our minimum Visual Studio version to 16.8 preview as well. The Net5 P8 SDK depends on having 16.8 MSBuild hence in order to develop roslyn after this change you will need 16.8 preview on your machine. 

I don't say this is a blocker, just as an FYI. It's been known for a while at a division level most of our repos would need to move to a preview VS for the end game of .NET5. We should tell the team though.

File: docs/contributing/Building, Debugging, and Testing on Windows.md
Comment (jasonmalinowski) [inline docs/contributing/Building, Debugging, and Testing on Windows.md:23]: Per @jaredpar's comments, this also bumps us to requiring 16.8 -- do we need to update line 18 accordingly?

Comment (sharwell) [reply docs/contributing/Building, Debugging, and Testing on Windows.md:23]: 💡 If we update the solution file appropriately, earlier versions of VS will actually know to report an error if you try to open it.

https://github.com/dotnet/roslyn/blob/255cb1e0543c986960da2bfdf265842c64fcc4a5/Roslyn.sln#L4

**Comment by 333fred** [comment]
I have validated building, testing, and F5'ing Roslyn.sln with 16.8p2. All seems to be working.

File: global.json
Comment (333fred) [inline global.json:10]: This seemed to work for the sdk to force a minimum of 16.8p2, so I'm giving it a try: https://github.com/dotnet/sdk/blob/master/global.json#L9-L10

Comment (jmarolf) [reply global.json:10]: this does not work on the public previews. This will need to be 16.8

I get this locally for 16.8 P2:
```
(NETCORE_ENGINEERING_TELEMETRY=InitializeToolset) Developer Command Prompt for VS 16.0 is not recent enough. Please upgrade to 16.9 or build from a plain CMD window
```

**Comment by 333fred** [comment]
I'm breaking the SCI and SRM updates out of this PR and into https://github.com/dotnet/roslyn/pull/47397 to unblock partner teams.

**Comment by 333fred** [comment]
@sharwell or @JoeRobich can you please look into the integration test failures? Feel free to push to this branch.

**Comment by JoeRobich** [comment]
hmm

image

**Comment by 333fred** [comment]
@JoeRobich or @sharwell, any progress here?

**Comment by 333fred** [comment]
I attempted to run the coreclr tests locally to figure out what is hanging. I wasn't able to see a hang locally, but I did see multiple test breakages from https://github.com/dotnet/runtime/issues/42234.

**Review by jmarolf — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]

File: azure-pipelines-integration.yml
Comment (jasonmalinowski) [inline azure-pipelines-integration.yml:21]: Do we still need this or is the Preview 2 queue now the main queue? We don't want to be permanently on the scouting queue or else we'd be get broken the moment it's updated.

Comment (jasonmalinowski) [reply azure-pipelines-integration.yml:21]: (fine to merge in this state if it's necessary but please fix ASAP as otherwise this is setting a trap to break things)

**Review by jasonmalinowski — APPROVED** [review]
Signing off to unblock the work, but please make sure the queue change to the integration test runs are rolled back as soon as practicable.

File: azure-pipelines-integration.yml
Comment (dibarbet) [reply azure-pipelines-integration.yml:21]: Preview2 is the main queue now, which is coincidentally why the integration tests aren't quite working
https://github.com/dotnet/core-eng/wiki/VS2019-Upgrade-Schedule

Comment (333fred) [reply azure-pipelines-integration.yml]: Moved back to the main pool.

File: azure-pipelines-integration.yml
Comment (sharwell) [inline azure-pipelines-integration.yml:31]: 😨 These are our _only_ tests for OOP.

Comment (sharwell) [reply azure-pipelines-integration.yml:31]: 64-bit testing covers the 99.9% customer scenario.

Comment (333fred) [reply azure-pipelines-integration.yml:31]: @tmat and @jasonmalinowski confirmed that we were not introducing any new product bugs here, only exposing existing issues. Given that this was also blocking other work, we opted to not hold this PR hostage to existing bugs that cannot be fixed at the same time.

Comment (jasonmalinowski) [reply azure-pipelines-integration.yml:31]: And in parallel with that analysis, an image upgrade to our machine pool also meant the builds were entirely broken for integration tests _unless_ we merged this PR to upgrade to Preview 8. So this quickly went from something we were doing with regret to something we had no alternative for.

Comment (sharwell) [reply azure-pipelines-integration.yml:31]: The integration tests were working in this pull request as of afbbec1 (10 September). I've requested this be treated as proof of a ship-blocking product regression until we can figure the reason why they stopped working again.

**Pull request merged**
